prov/verbs/src/verbs_eq.c

```diff
@@ -428,11 +445,18 @@ fi_ibv_eq_xrc_rej_event(struct fi_ibv_eq *eq, struct rdma_cm_event *cma_event)
 	enum fi_ibv_xrc_ep_conn_state state;
 
 	ep = container_of(fid, struct fi_ibv_xrc_ep, base_ep.util_ep.ep_fid);
-	state = ep->conn_state;
+	if (ep->magic != VERBS_XRC_EP_MAGIC) {
```

Comment:
What purpose is the magic value serving in this context?

Reply:
It was added as a sanity check intended to catch bad state. It clearly should not happen in normal operation, but if it did while we are trying to scale up it would be nice to know. I can take it out if you prefer, make it an assert, or leave it as is.

This goes back to my concern that with multiple EP or auto progress. The way the CM processing  is done: i.e. a lock is taken to read the event, then released, then taken again around protecting state in the handler, there is nothing that prevents multiple EP form processing events associated with the same ID (if they were to be delivered by  the RDMA CM).

Reply:
I should have been more clear about my question. Typically, I've seen magic values to ensure that structs were not corrupted by bad copies of data, or that structs were not reused without going through the proper allocation and deallocation functions. What is this check looking for? bad state of the connection manager, bad state from allocation/deallocation,etc...

Reply:
It is attempting to catch any races or bad state in the CM manager logic under load. The release of an EP will destroy the CM IDs (and clear the magic) so we should never see this condition if all locking is working correctly. As you can probably tell, I'm not confident that all the bases are covered and I'd like notification if not.

If the delivered development builds have debug enabled, then an assert would probably be good enough here, but if not, I think it is useful to catch this kind of stuff for now. In general, the provider assumes everything is working great, and I think that is a little overly optimistic at this point.